pkg/runner/runner.go

```diff
@@ -135,31 +143,65 @@ func selectPlan(spec *bundle.Spec) bundle.Plan {
 	return bundle.Plan{}
 }
 
-func selectParameters(plan bundle.Plan) bundle.Parameters {
+func selectParameters(plan bundle.Plan) (bundle.Parameters, error) {
+	schemaPlan, err := bundle.ConvertPlansToSchema([]bundle.Plan{plan})
+	if err != nil {
+		log.Errorf("Error converting bundle plans to JSON Schema: %v", err)
+		return nil, err
+	}
+	planSchema := schemaPlan[0].Schemas
+	schemaParams := planSchema.ServiceInstance.Create["parameters"]
 	params := bundle.Parameters{}
 	for _, param := range plan.Parameters {
-		var paramDefault string
-		var paramInput string
+		var paramDefault interface{}
 		if param.Default != nil {
-			paramDefault = param.Default.(string)
+			paramDefault = param.Default
 		}
 		check := 1
 		for check < 2 {
+			var paramInput string
 			fmt.Printf("Enter value for parameter [%v], default: [%v]: ", param.Name, paramDefault)
 			fmt.Scanln(&paramInput)
 			if paramInput == "" {
-				paramInput = paramDefault
+				switch paramDefault.(type) {
```

Comment:
Haven't seen this type of expression before. Is this just a cast?

Reply:
This is actually a special thing in Go... you cannot evaluate `paramDefault.(type)` outside of a switch statement. It helps for doing different things based on input.